src/Processors/Formats/Impl/AvroRowInputFormat.cpp

```diff
@@ -729,57 +777,62 @@ static LRUCache<std::string, ConfluentSchemaRegistry>  schema_registry_cache(SCH
 
 static std::shared_ptr<ConfluentSchemaRegistry> getConfluentSchemaRegistry(const FormatSettings & format_settings)
 {
-    const auto & base_url = format_settings.avro.schema_registry_url;
+    /// proton: starts
+    const auto & base_url = format_settings.schema.kafka_schema_registry_url.empty() ? format_settings.avro.schema_registry_url : format_settings.schema.kafka_schema_registry_url;
+    const auto & credentials = format_settings.schema.kafka_schema_registry_credentials;
     auto [schema_registry, loaded] = schema_registry_cache.getOrSet(
-        base_url,
-        [base_url]()
+        base_url + credentials,
+        [base_url, credentials]()
```

Comment:
if this is a sync call, we can ref capture here for base_url and credentials ? like 

```
[&] {
}
```